Feature/chapter 6 - System Maintenance tests

JBartscher opened the pull request:
Add all tests for chapter 6. I checked every test case and could ensure they are passable/failable. However, with test 6.2.6 & 6.2.7 I was not able to apply the remedial measures.

JBartscher committed:
Add tests for chapter 6

For test 6.1.11 & 6.1.12 the remediation could not be tried successfully. The test on the other hand does what it is supposed to do. Therefore we will see in the later implementation for the ecosystem if we can find a solution or if it is just a local problem with my vagrant machine.

JBartscher committed:
Ensure 6.1 and 6.2 test can fail and pass

Test of all tests via they can fail and pass.

robertauer commented on 6-System-Maintenance/6.2-User-and-Group-Settings.bats:
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bats
+
+@test "6.2.1 Ensure accounts in /etc/passwd use shadowed passwords (Automated)" {
+  run "$BATS_TEST_DIRNAME"/6.2.1.sh
```
This script only holds one line. I think you can run this line here directly.

robertauer commented on 6-System-Maintenance/6.2-User-and-Group-Settings.bats:
```diff
@@ -0,0 +1,113 @@
+
+@test "6.2.2 Ensure password fields are not empty (Automated)" {
+  # must be run as root, otherwise files that are not accessible will be erroneously specified as a finding.
+  run sudo "$BATS_TEST_DIRNAME"/6.2.2.sh
```
- This script only holds one line. I think you can run this line here directly.
- I think we can assume all the scripts will be executed as root user, so no sudo needed.

robertauer commented on 6-System-Maintenance/6.2.3.sh:
```diff
@@ -0,0 +1,9 @@
+  grep -q -P "^.*?:[^:]*:$i:" /etc/group
+  if [ $? -ne 0 ]; then
+    echo "Group $i is referenced by /etc/passwd but does not exist in
+/etc/group"
```
I think this belongs at the end of line 6

robertauer commented on 6-System-Maintenance/6.2.4.sh:
```diff
@@ -0,0 +1,11 @@
+
+  if [ ! -d "$dir" ]; then
+    echo "User: \"$user\" home directory: \"$dir\" does not
+  exist."
```
I think this belongs at the end of line 7

robertauer commented on 6-System-Maintenance/6.2.4.sh:
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+awk -F: '($1!~/(halt|sync|shutdown|nfsnobody)/ &&
+$7!~/^(\/usr)?\/sbin\/nologin(\/)?$/ && $7!~/(\/usr)?\/bin\/false(\/)?$/) {print $1 " " $6 }' /etc/passwd | while read -r user dir; do
```
I think this belongs at the end of line 3. Keep in mind to put a space between the `&&` and the `$7!...`

robertauer commented on 6-System-Maintenance/6.2.6.sh:
```diff
@@ -0,0 +1,20 @@
+      "$dirperm" | cut -c10
+    )" != "-" ]; then
+      echo "User: \"$user\" home directory: \"$dir\" has permissions:
+\"$(stat -L -c "%a" "$dir")\""
```
I think this belongs at the end of line 15

robertauer commented on 6-System-Maintenance/6.2.7.sh:
```diff
@@ -0,0 +1,20 @@
+          "$fileperm" | cut -c9
+        )" != "-" ]; then
+          echo "User: \"$user\" file: \"$file\" has permissions:
+\"$fileperm\""
```
I think this belongs at the end of line 14

robertauer commented on 6-System-Maintenance/6.2.8.sh:
```diff
@@ -0,0 +1,19 @@
+    if [ ! -h "$file" ] && [ -f "$file" ]; then
+      if stat -L -c "%A" "$file" | cut -c4-10 | grep -Eq '[^-]+'; then
+        echo "FAILED: User: \"$user\" file: \"$file\" exists with
+permissions: \"$(stat -L -c "%a" "$file")\", remove file or excessive
+permissions"
```
I think this belongs at the end of line 10

robertauer commented on 6-System-Maintenance/6.2.8.sh:
```diff
@@ -0,0 +1,19 @@
+permissions"
+      else
+        echo "WARNING: User: \"$user\" file: \"$file\" exists with
+permissions: \"$(stat -L -c "%a" "$file")\", remove file unless required"
```
I think this belongs at the end of line 14

robertauer commented on 6-System-Maintenance/6.2-User-and-Group-Settings.bats:
```diff
@@ -0,0 +1,113 @@
+  [ "$status" -eq 0 ]
+  [[ "$output" == "" ]]
+
+  run "$BATS_TEST_DIRNAME"/6.2.17.sh
```
This script only holds one line. I think you can run this line here directly.

JBartscher committed:
Apply requested changes from code review

Test 6.2.1, 6.2.2 and 6.2.17 are no longer stored in a separate file, as they were only one line long.
Reformatting shell scripts to not break long output lines.
